overlord/snapshotstate/snapshotmgr.go

```diff
@@ -371,6 +372,23 @@ func undoRestore(task *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+func doTaskCleanup(task *state.Task, tomb *tomb.Tomb) error {
+	return cleanupRestore(task, tomb)
+}
+
+func doCleanup(task *state.Task, tomb *tomb.Tomb) error {
+	for _, t := range task.WaitTasks() {
```

Comment:
This is accessed without acquiring the state lock (there are some unit test failures)

Reply:
Thanks, you were correct indeed!